### PR TITLE
add checkErrorHandler

### PR DESCRIPTION
**Pull request by shileiwill**
AUTO-9015

Add checkErrorHandler in contracts/src/v0.8/automation/interfaces/StreamsLookupCompatibleInterface.sol

All new contracts can add this checkErrorHandler function so they can have the streams lookup errors.

### CODE REVIEW

File: contracts/src/v0.8/automation/interfaces/StreamsLookupCompatibleInterface.sol [hunk 1]
```diff
@@ -17,4 +17,17 @@ interface StreamsLookupCompatibleInterface {
     bytes[] memory values,
     bytes memory extraData
   ) external view returns (bool upkeepNeeded, bytes memory performData);
+
+  /**
+   * @dev this is a new, optional function in v2.1. It is meant to surface streams lookup errors.
```
Comment (infiloop2) [inline contracts/src/v0.8/automation/interfaces/StreamsLookupCompatibleInterface.sol:22]: nit.

- use @notice as above function
- `this is a new, optional function in v2.1.` -> `this is a new, optional function in streams lookup`. (Streams lookup is independent of v2.1 and will be supported in future versions)

File: contracts/src/v0.8/automation/dev/MercuryRegistry.sol [hunk 2]
```diff
@@ -144,6 +144,14 @@ contract MercuryRegistry is ConfirmedOwner, AutomationCompatibleInterface, Strea
...
   }
 
+  function checkErrorHandler(
+    uint256 errCode,
+    bytes memory extraData
+  ) external view override returns (bool upkeepNeeded, bytes memory performData) {
+    // dummy function, just return true and extraData
+    return (true, extraData);
+  }
+
```
Comment (infiloop2) [inline contracts/src/v0.8/automation/dev/MercuryRegistry.sol:154]: default implementation should be false to keep existing behaviour

return (false, new bytes(0))

Also you might be able to define this within the interface itself without requiring all contract code to change

File: contracts/src/v0.8/tests/VerifiableLoadLogTriggerUpkeep.sol [hunk 3]
```diff
@@ -116,4 +116,12 @@ contract VerifiableLoadLogTriggerUpkeep is VerifiableLoadBase, StreamsLookupComp
     bytes memory performData = abi.encode(values, extraData);
     return (true, performData);
   }
+
+  function checkErrorHandler(
```
[the remark this answers is not in the document]
Comment (shileiwill) [reply contracts/src/v0.8/tests/VerifiableLoadLogTriggerUpkeep.sol:120]: Yes, this is a test contract used by automation, will do the migration, tracked in https://smartcontract-it.atlassian.net/browse/AUTO-9028

File: contracts/src/v0.8/automation/dev/MercuryRegistry.sol (hunk 2 above)
Comment (shileiwill) [reply contracts/src/v0.8/automation/dev/MercuryRegistry.sol:154]: well, just checked : Functions in interfaces cannot have an implementation.

**Pull request merged**